core/src/HIP/Kokkos_HIP_Error.hpp

```diff
@@ -28,9 +28,10 @@ inline void hip_internal_safe_call(hipError_t e, const char* name,
   switch (e) {
     case hipSuccess: break;
     case hipErrorInvalidValue:
-    case hipErrorOutOfMemory:
 #if (HIP_VERSION_MAJOR >= 7)
     case hipErrorMemoryAllocation:
+#else
+    case hipErrorOutOfMemory:
 #endif
```

Comment:
```suggestion
    case hipErrorOutOfMemory:
```
Because the error name and explanatory string cannot distinguish the two and as far as I can tell the `hipError_t ::hipErrorOutOfMemory` enumerator is not deprecated.
https://github.com/ROCm/rocm-systems/blob/952d1dabe252450eda9e0a223aa178f4fa0fc695/projects/clr/hipamd/src/hip_error.cpp#L46-L52
https://github.com/ROCm/rocm-systems/blob/952d1dabe252450eda9e0a223aa178f4fa0fc695/projects/clr/hipamd/src/hip_error.cpp#L201-L208

Reply:
The deprecation (`hipErrorMemoryAllocation`->`hipErrorOutOfMemory`) is in https://github.com/ROCm/rocm-systems/commit/6a881a647ce6cab9fc247ecc436723154ecdd0cd.